### PR TITLE
New testing api

### PR DESCRIPTION
**Pull request by DominikWolek**
- Updated test due to new testing api
- Replaced Travis with CircleCI

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      - image: membrane/bionic-membrane:latest
+    environment:
+      MIX_ENV: test
+      PKG_CONFIG_PATH: '/root/app/ext/'
+    working_directory: '~/app'
+
+    steps:
+      - checkout
+      - run: update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-7 100
+      - run: apt-get install -qq libmp3lame-dev
```
Comment (Hajto) [inline .circleci/config.yml]: Maybe this could be baked into the image?

File: mix.exs [hunk 2]
```diff
@@ -63,7 +63,11 @@ defmodule Membrane.Element.Lame.Mixfile do
   defp deps do
     [
       {:ex_doc, "~> 0.19", only: :dev, runtime: false},
-      {:membrane_core, "~> 0.3.0"},
+      {:membrane_core,
+       github: "membraneframework/membrane-core",
+       branch: "new-testing-api",
+       only: [:test, :dev],
```
Comment (Hajto) [inline mix.exs]: Why only :test and :dev?

File: test/integration/integration_test.exs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  alias Membrane.Element
+
+  def make_pipeline(in_path, out_path) do
+    Pipeline.start_link(%Pipeline.Options{
+      elements: [
+        file_src: %Element.File.Source{chunk_size: 4096, location: in_path},
+        encoder: Element.Lame.Encoder,
+        sink: %Element.File.Sink{location: out_path}
+      ],
+      links: %{
```
Comment (Hajto) [inline test/integration/integration_test.exs]: Are those links needed?

File: test/integration/integration_test.exs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  test "Encode raw samples" do
+    in_path = "../fixtures/input.pcm" |> Path.expand(__DIR__)
+    out_path = "/tmp/output-lame.mp3"
+    on_exit(fn -> File.rm(out_path) end)
+
+    assert {:ok, pid} = make_pipeline(in_path, out_path)
+
+    assert Pipeline.play(pid) == :ok
+    assert_end_of_stream(pid, :sink, :input, 300)
```
Comment (Hajto) [inline test/integration/integration_test.exs]: Shouldn't you assert here wether output and reference files are equal or something?

File: .circleci/config.yml (hunk 1 above)
Comment (DominikWolek) [reply .circleci/config.yml]: It doesn't make build much slower

File: mix.exs [hunk 5]
```diff
@@ -64,10 +64,7 @@ defmodule Membrane.Element.Lame.Mixfile do
     [
       {:ex_doc, "~> 0.19", only: :dev, runtime: false},
       {:membrane_core,
-       github: "membraneframework/membrane-core",
-       branch: "new-testing-api",
-       only: [:test, :dev],
-       override: true},
+       github: "membraneframework/membrane-core", branch: "new-testing-api", override: true},
```
Comment (Hajto) [inline mix.exs]: Wouldn't master suffice?

**Pull request merged**